### PR TITLE
Issues#13

### PR DESCRIPTION
**Pull request by at-tuannd**
- create upload in ckfinder.
- add, edit, delete category tour.
- order category tour ajax.

### CODE REVIEW

File: app/Http/Controllers/Admin/CategoryTourController.php [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $listArea=$this->areaRepository->all();
```
Comment (at-vietpb) [inline app/Http/Controllers/Admin/CategoryTourController.php]: Never get all data from database.

File: app/Http/Controllers/Admin/CategoryTourController.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+     * Remove the specified resource from storage.
+     *
+     * @param int $id [descriptions]
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        if (Auth::user()->level==config('constants.level_admin') || Auth::user()->level!=config('constants.level_admin') && $this->categoryTourRepositories->countCategoryTour($id, Auth::user()->id)>config('constants.countdestroycategorytour')) {
+            $this->cateRepositories->delete($id);
```
Comment (at-vietpb) [inline app/Http/Controllers/Admin/CategoryTourController.php]: What happened with sub-categories and Tours belongs to this category and sub categories?

File: app/Http/Controllers/Admin/CategoryTourController.php [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+     * @param Illuminate\Http\Request $request [description]
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function postOrder(Request $request)
+    {
+        if ($request::ajax()) {
+            $id=$request::input('idcategorytour');
+            $order=$request::input('order');
+            return $this->cateRepositories->oder($id, $order);
```
Comment (at-vietpb) [inline app/Http/Controllers/Admin/CategoryTourController.php]: `oder` OR `order`?

File: app/Http/Requests/CategoryTourEditRequest.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'name_category_tour'=>'required',
```
Comment (at-vietpb) [inline app/Http/Requests/CategoryTourEditRequest.php]: In add new request you have rule `unique:category_tours,name_category_tour`
Why on edit request you don't add this rule?

File: app/Http/Controllers/Admin/CategoryTourController.php (hunk 3 above)
Comment (at-tuannd) [inline app/Http/Controllers/Admin/CategoryTourController.php]: yes. thank you

File: app/Http/Controllers/Admin/CategoryTourController.php (hunk 2 above)
Comment (at-tuannd) [inline app/Http/Controllers/Admin/CategoryTourController.php]: Yes. I have dealt in migrate 2017_01_09_033355_create_category_tours_table.php

File: app/Http/Requests/CategoryTourEditRequest.php (hunk 4 above)
Comment (at-tuannd) [inline app/Http/Requests/CategoryTourEditRequest.php]: yes. thank you

**Pull request merged**